src/language-js/needs-parens.js

```diff
@@ -561,6 +579,20 @@ function needsParens(path, options) {
 
     case "ArrowFunctionExpression":
       switch (parent.type) {
+        case "PipelineTopicExpression":
+          return !!(node.extra && node.extra.parenthesized);
+
+        case "BinaryExpression":
+          if (node.extra && node.extra.parenthesized) {
+            return (
+              parent.operator === "|>" ||
+              (parent.operator !== "|>" &&
+                node.body.type === "BinaryExpression" &&
+                node.body.operator === "|>")
+            );
+          }
+          return false;
```

Comment:
This leads to a bug:

**Prettier pr-6319**
[Playground link](https://deploy-preview-6319--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAKVAPAlAAgLwB8O2OA1DgOQUDcIANCBAA4wCW0AzsqAIYBOfCAHcACvwRcUPADZCeATy4MARnx5gA1nBgBlJutZQA5shh8ArnAYALGAFtpAdWut4HfWDg6Jr1gDdXeWRwDiUQQw44PhgRNSM7HmQAMxlIhgArDgwAITVNbR0eOzgAGUM4ZNSrEEyMHUMjaTgARXMIeErpNJB9Pki+YJh5JjgOMD5WFnoeidhHVgATGGtkAA4ABgYmQUjHNSZg7dGovwqGAEc2+FjmSRAeDgBaKDg4BbfpvjhL1i-YnniiSQKS61UidlYpgsYIaTVa7QqwKqDBgPGU8yWKyQACYUWpWNIGgBhCB2BLBUYAVmm5kiABU0ZIQd0-JYAJJQd6wHTjSYwACCnJ0QyanUiAF9xUA)
```sh
--parser typescript
```

**Input:**
```tsx
((x) => x) + '';
```

**Output:**
```tsx
(x) => x + "";

```